reactive/kotlinx-coroutines-rx2/src/RxObservable.kt

```diff
@@ -164,12 +164,10 @@ private class RxObservableCoroutine<T: Any>(
                          * Such behaviour is inconsistent, leads to silent failures and we can't possibly know whether
                          * the cause will be handled by onError (and moreover, it depends on whether a fatal exception was
                          * thrown by subscriber or upstream).
-                         * To make behaviour consistent and least surprising, we always handle fatal exceptions
-                         * by coroutines machinery, anyway, they should not be present in regular program flow,
-                         * thus our goal here is just to expose it as soon as possible.
+                         * To make behaviour consistent and least surprising, we always deliver fatal exceptions to the
+                         * RX global exception handler.
                          */
-                        subscriber.tryOnError(cause)
-                        if (!handled && cause.isFatal()) {
+                        if (!subscriber.tryOnError(cause) || (!handled && cause.isFatal())) {
```

Comment:
It seems that the `handled` flag is redundant here after the earlier check for `CancellationException`, since it could only be handled in the current implementation if it was a `CancellationException`. There doesn't seem to be any special handling of fatal exceptions by the coroutine. However, I have left that logic as it was here.